scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -211,6 +219,46 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         i
       }
       val next = if (nextPos != -1) scannerTokens(nextPos) else null
+
+      val currentIndent = {
+        if (curr.is[Whitespace]) {
+          -1
+        } else {
+          var i = currPos - 1
+          var ind = 0
+          var commentLF = false
+          def isWhitespace(token: Token): Boolean =
+            token.is[Space] || token.is[Tab] || token.is[Comment]
+          while (i >= 0 && isWhitespace(scannerTokens(i))) {
```

Comment:
1. do you need to test for `commentLF`?

2 .opt: implement as `@tailrec def`, should be just as fast but a bit more readable.

Reply:
1. removed for now.
2. implemented :+1: